pytests/test_datetime.py

```diff
@@ -0,0 +1,166 @@
+        py39 = False
+        start_at = datetime.now(tz=timezone(timedelta(hours=-8)))
+
+    wc = TumblingWindowConfig(length=timedelta(seconds=2), start_at=start_at)
+
+    # The dates should always be equivalent
+    assert start_at == wc.start_at
+    # But if we are working with fixed offsets in python<3.9,
+    # we convert everything to UTC, so the tzinfo won't be the same.
+    if py39:
+        assert start_at.tzinfo == wc.start_at.tzinfo
+
+
+def test_naive_datetime():
+    """
+    Test that naive datetimes are converted to UTC, assuming local datetime.
+    """
+    start_at = datetime.now()
+    wc = TumblingWindowConfig(length=timedelta(seconds=2), start_at=start_at)
+    assert start_at.astimezone(timezone.utc) == wc.start_at
+
+
+def test_fold_window_tzinfo():
+    def gen(ws, wi, rs):
+        yield from [
+            (None, {"user": "a", "type": "login"}),
+            (None, {"user": "a", "type": "post"}),
+            (None, {"user": "a", "type": "post"}),
+            (None, {"user": "b", "type": "login"}),
+            (None, {"user": "a", "type": "post"}),
+            (None, {"user": "b", "type": "post"}),
+            (None, {"user": "b", "type": "post"}),
+        ]
+
+    def extract_id(event):
+        return (event["user"], event)
+
+    def build():
+        return defaultdict(lambda: 0)
+
+    def count(results, event):
+        results[event["type"]] += 1
+        return results
+
+    def build_flow(flow, window_config, out):
+        flow.input("inp", ManualInputConfig(gen))
+        flow.map(extract_id)
+        flow.fold_window("sum", clock_config, window_config, build, count)
+        flow.capture(TestingOutputConfig(out))
+
+    # In python>=3.9 we have the ZoneInfo package, otherwise we use
+    # a fixed offset as the timezone info
+    try:
+        from zoneinfo import ZoneInfo
+
+        start_at = datetime.now(ZoneInfo("US/Pacific"))
+        # Another representation of the same date
+        start_at_1 = start_at.astimezone(ZoneInfo("Europe/Rome"))
+    except ImportError:
+        start_at = datetime.now(tz=timezone(timedelta(hours=-8)))
+        # Another representation of the same date
+        start_at_1 = start_at.astimezone(tz=timezone(timedelta(hours=2)))
+
+    assert start_at == start_at_1
+
+    clock_config = TestingClockConfig(item_incr=timedelta(seconds=4), start_at=start_at)
+
+    # Now try the flow with the two different dates, we should have the same result:
+    window_config_1 = TumblingWindowConfig(
+        length=timedelta(seconds=10), start_at=start_at
+    )
+    window_config_2 = TumblingWindowConfig(
+        length=timedelta(seconds=10), start_at=start_at_1
+    )
+
+    out_1 = []
+    flow_1 = Dataflow()
+    build_flow(flow_1, window_config_1, out_1)
+
+    out_2 = []
+    flow_2 = Dataflow()
+    build_flow(flow_2, window_config_2, out_2)
+
+    run_main(flow_1)
+    run_main(flow_2)
+
+    assert len(out_1) == len(out_2)
+    assert out_1 == out_2
```

Comment:
this is a regression test (maybe should add a comment about it). It is testing a behaviour that is supposed to be always correct, but if you run this test using the `pyo3-chrono` integration we used before, this test fails because even if `start_at` and `start_at_1` represents the same date in python, the incorrect conversion in `pyo3-chrono` made the two flow run with a different `start_at` parameter, and gave two different results.